src/lib/passport/fib.ts

```diff
@@ -41,28 +42,31 @@ const strategy = new OAuth2Strategy(
     _profile: any,
     done: any,
   ) => {
-    const { username } = await getFibUser(accessToken);
+    const usernameHash = await getFibUserHash(accessToken);
     const expirationDate = createExpirationDate(params.expires_in);
 
     const user: UserToken = {
       accessToken,
       refreshToken,
       expirationDate,
-      username: username,
+      usernameHash: usernameHash,
     };
 
     return done(null, user);
   },
 );
 
-async function getFibUser(accessToken: string): Promise<userFibApi> {
+async function getFibUserHash(accessToken: string): Promise<string> {
   const response = await fetch(`${API_URL}/jo?format=json`, {
     headers: {
       Authorization: `Bearer ${accessToken}`,
     },
   });
 
-  return (await response.json()) as userFibApi;
+  const { username } = (await response.json()) as userFibApi;
+
+  // IMPROVEMENT: Instead of only hashing the username, we could encrypt it for extra security.
```

Comment:
Not sure how this would make it more secure. Hashing already obfuscates the usernames.

Reply:
Agreed to include a salt.